p2pserver/message/msg_pack/msg_pack.go

```diff
@@ -208,12 +208,15 @@ func NewVersionPayload(n p2pnet.P2P, isCons bool, height uint32) mt.VersionPaylo
 		Version:      n.GetVersion(),
 		Services:     n.GetServices(),
 		SyncPort:     n.GetSyncPort(),
-		ConsPort:     n.GetConsPort(),
 		Nonce:        n.GetID(),
 		IsConsensus:  isCons,
 		HttpInfoPort: n.GetHttpInfoPort(),
 	}
-
+	if config.DefConfig.P2PNode.DualPortSupport {
+		vpl.ConsPort = n.GetConsPort()
```

Comment:
why not set ConsPort right on p2p module initialisation, so n.GetConsPort() returns expected answer?

Reply:
good idea, logic added in netserver